weston/weston-init

```diff
@@ -28,11 +28,72 @@ startup-animation=fade
 
 IEOF
 
+WPE_WESTON_OUTPUT_MAX_W=${WPE_WESTON_OUTPUT_MAX_W:-0}
+WPE_WESTON_OUTPUT_MAX_H=${WPE_WESTON_OUTPUT_MAX_H:-0}
+
+# Limit resolution if necessary
+if [ "$WPE_WESTON_OUTPUT_MAX_W" -gt 0 -o "$WPE_WESTON_OUTPUT_MAX_H" -gt 0 ]; then
+    for outp in /sys/class/drm/card*-*; do
+        # no outputs
+        if [ ! -d "$outp" ]; then
+            break
+        fi
+        # skip if not enabled
+        if [ "$(cat ${outp}/enabled)" = "disabled" ]; then
+            continue
+        fi
+        # get the output name
+        outn=${outp#*-}
+        # check the modes
+        moden=1
+        foundres=
+        for mode in $(cat "${outp}/modes"); do
+            wres=${mode%x*}
+            hres=${mode#*x}
+            # skip interlaced modes and other junk
+            if ! [ "$wres" -eq "$wres" -a "$hres" -eq "$hres" ] 2> /dev/null; then
+                moden=$(($moden + 1))
+                continue
+            fi
+            if [ "$WPE_WESTON_OUTPUT_MAX_W" -gt 0 ]; then
+                if [ "$wres" -gt "$WPE_WESTON_OUTPUT_MAX_W" ]; then
+                    moden=$(($moden + 1))
+                    continue
+                fi
+            fi
+            if [ "$WPE_WESTON_OUTPUT_MAX_H" -gt 0 ]; then
+                if [ "$hres" -gt "$WPE_WESTON_OUTPUT_MAX_H" ]; then
+                    moden=$(($moden + 1))
+                    continue
+                fi
+            fi
+            # found an acceptable res
+            foundres="$mode"
+            break
+        done
+        # found a res to use and it's non-preferred, override in weston
+        if [ -n "$foundres" -a "$moden" -gt 1 ]; then
+            cat << IEOF >> /weston.ini
+[output]
```

Comment:
Maybe leave extra empty lines so the resulting weston.ini has some separation?  I think that in its current form, there's no separation from previous blocks.

What happens if there are multiple [output] sections (i.e., a previous one before you add this one), do they get merged properly, or will the new section will clobber completely the old one?  (Or, ideally, the second section doesn't clobber completely but overrides the mode for that output if the output is specified in a previous [output] section)?

Reply:
weston.ini explicitly allows multiple output sections, it's how you configure multiple outputs

Reply:
Ref: https://github.com/Igalia/balena-wpe/blob/4a46b3fce43a69d0f390cefb13e1ab4ea670207f/weston/weston-init

Right now the weston.ini is created on the fly. The resulting new weston.ini contains only one [output] section. No way to get it duplicated. I think.

Reply:
there is no way to get a dupe, as it creates up to 1 output section per physical output, and there is no output section otherwise